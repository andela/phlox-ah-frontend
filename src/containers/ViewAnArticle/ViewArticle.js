import React, { Component } from 'react';
import Avatar from 'react-avatar';
import PropTypes from 'prop-types';
import Moment from 'moment';
import { Row } from 'react-materialize';
import './ViewArticle.scss';
import { connect } from 'react-redux';
import ReactHtmlParser from 'react-html-parser';
import { viewArticle } from '../../requests/ArticleRequests';

/**
 * @class ViewAnArticle
 * @extends {Component}
 */
class ViewArticle extends Component {
  /**
   * @constructor function
   * @param {*} props React properties
   */
  constructor() {
    super();
    this.state = {
      success: false, loading: false, failure: false, article: []
    };
  }

  /**
   * @description - This method runs whenever the redux state changes
   * @returns {object} state
   * @param {object} props
   * @param {object} state
   */
  static getDerivedStateFromProps(props, state) {
    if (props.success) {
      const { success, loading, article } = props;
      return { success, loading, article };
    }
    if (props.failure) {
      const {
        success, loading, article, failure
      } = props;
      return {
        success, loading, article, failure
      };
    }
    return state;
  }

  /**
   * @description - This method runs whenever the initial component renders
   * @returns {object} state
   * @param {object} props
   */
  componentDidMount() {
    this.props.viewArticle(this.props.match.params.articleslug);
  }

  /**
   *
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   */
  render() {
    const { success, article, failure } = this.state;
    return (
    <div className="MainWrapper2">
        <div className="container">
        <Row className="containerRow">
            <div className="col s12 mainContainer ">
            <Row>
                {
                    failure
                    && <div className="center failure">
                         <h1><i className="fas fa-exclamation-triangle"></i></h1>
                         <h1>404</h1>
                         <h2 className="capitalize">We couldnâ€™t find this page.</h2>
                      </div>
                }
                {/* DUMMY DIV */}
                { !success
                  ? <div className={failure && 'hide'}>
                <div className="col s12 l6 lighten-5 img-div">
                <div className="dummyPicture"></div>
                </div>
                <div className="col s12 l6 lighten-5">
                <div className="dummyRight"><div></div><div></div><div></div></div>
                </div>
                <div className="col s12 article-body">
                <div className="dummyBody"><div></div><div></div><div></div></div>
                </div>
                </div>
                //   REAL DIV
                  : <div>
                    <div className="col s12 l6 img-div">
                        <img src={article.imgUrl}/>
                    </div>
                    <div className="col s12 l6">
                        <div>
                        <h4 className="capitalize">{article.title}</h4>
                        <p className="articleDescription">{article.description}</p>
                        <Row className="margin-top-10">
                            <div className="col s8">
                            <Row className="valign-wrapper">
                                <div className="col s4 m3 l4">
                                {
                                    !article.User.Profile || !article.User.Profile.profileImage ? <Avatar name={article.User.username} size="75" round={true} />
                                      : <img className="profileImage" src={article.User.Profile.profileImage}/>
                                }
                                </div>
                                <div className="col s8 m9 l8">
                                <span className="writer capitalize">{article.User.username}</span><br/>
                                <span className="date-written capitalize">{Moment.duration(article.createdAt, 'hours').humanize() }</span><br/>
                                <span className="readTime">{article.readTime} Minutes Read</span>
                                </div>
                            </Row>
                            </div>
                            <div className="col s4">
                            <button
                                className="btn waves-effect waves-light followButton"
                                type="submit"
                                name="action">Follow
                            </button>
                            </div>
                        </Row>
                        <div className="center-align activity-icons">
                            <div className="col s3"><i className="fas fa-thumbs-up likeButton liked-unliked"></i> {article.likes.length}</div>
                            <div className="col s3"><i className="fas fa-thumbs-down dislikeButton"></i> 97</div>
                            <div className="col s3"><i className="fas fa-bookmark bookmarkButton"></i></div>
                            <div className="col s3"><i className="fas fa-share-alt shareButton"></i></div>
                        </div>
                        </div>
                    </div>
                    <div className="col s12 article-body">
                        {
                        ReactHtmlParser(article.body)
                        }
                    </div>
                    <div className="col l4 s12 bold tag-div">
                        {<a className="red-text"href="#">Report Article</a>}
                    </div>
                    </div>
                }
            </Row>
            </div>
        </Row>
        </div>
    </div>
    );
  }
}

ViewArticle.propTypes = {
  viewArticle: PropTypes.func.isRequired,
  loading: PropTypes.bool,
  success: PropTypes.bool,
  failure: PropTypes.bool,
  articles: PropTypes.object,
  match: PropTypes.object
};

const mapStateToProps = state => ({
  loading: state.article.loading,
  success: state.article.success,
  failure: state.article.failure,
  article: state.article.article,
});

export default connect(mapStateToProps, { viewArticle })(ViewArticle);
