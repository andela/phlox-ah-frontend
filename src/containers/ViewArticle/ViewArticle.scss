@import '../../styles/variables';

.MainWrapper2 {
  background-color: #f8f9fa;
  .dummyBody {
    div {
      width: 100%;
      height: 50px;
      background-color: #e5e5e5;
      margin-top: 5px;
    }
  }
  .failure{
    h1{
      border-bottom: none;
      font-weight:900;
      font-size:100px;
      margin:10px 0px;
      padding:0px;
      color:#B2B2B2;
    }
    i{
      font-size:50px;
      color:#EF5350;
    }
    h2 {
      font-size: 30px;
    }
  }
  .container {
    width: 90%;
    margin-top: 2px;
    .containerRow {
      padding-top: 80px;
      background-color: #ffffff;
    }
    .mainContainer {
      padding-left: 20px;
      padding-right: 20px;
      margin-bottom: 50px;
    }
    .dummyRight {
      div {
        width: 100%;
        height: 100px;
        background-color: #e5e5e5;
        margin-top: 5px;
      }
    }
    .img-div {
      img {
        width: 100%;
        height: 350px;
      }
      .dummyPicture {
        width: 100%;
        height: 350px;
        background-color: #e5e5e5;
      }
    }
    h4 {
      font-weight: bold;
      font-size: 20px;
    }
    p {
      margin-bottom: 15px;
      text-align: justify;
    }
    .profileImage {
      width: 50px;
      height: 50px;
      border-radius: 50%;
    }
    .followButton {
      background-color: #e5e5e5;
      color: #5e5f63;
      border: 1px solid #5e5f63;
    }
    .editButton {
      background-color: #3B8CA1;
      color: white;
      border: none;
    }
    p.articleDescription {
      height: 150px;
      padding-top: 3%;
    }
    span.red-text {
      float: right;
      text-decoration: none;
      cursor: pointer;
    }
    a.btn.waves-effect.waves-light.editButton {
      position: relative;
      right: 100%;
      bottom: 30px;
    }
    .margin-top-10 {
      margin-top: 10px;
    }
    .articleDescription {
      font-size: 0.85em;
      color: #6f6a6a;
    }
    .writer {
      font-weight: bold;
    }
    .activity-icons {
      i {
        cursor: pointer;
      }
      .active{
        color: #3b8ca1;
      }
      color: #5e5f63;
      .bookmarked {
        color: #3B8CA1;
      }
      .not-bookmarked {
        color: #5E5F63;
      }
      .no-bookmark {
        cursor: default;
      }
    }
    .liked-unliked {
      color: #3b8ca1;
    }
    .article-body, .tag-div {
      margin-top: 50px;
      text-align: justify;
      width: 80%;
      position: relative;
      left: 10%;
      right: 10%;
      @media #{$desktop} {
        width: 90%;
      }
      @media #{$tablet} {
        width: 94%;
        left: 0%;
        right: 0%;
      }
      @media #{$mobile} {
        width: 100%;
        left: 0%;
        right: 0%;
      }
    }
    .bold {
      font-weight: bold;
    }
    @media only screen and (min-width: 601px) {
      .hide-on-small-only {
        display: block;
      }
      .mainContainer {
        min-height: 800px;
        padding-left: 50px;
        padding-right: 50px;
      }
      .followButton {
        background-color: #e5e5e5;
        color: #5e5f63;
        border: 1px solid #5e5f63;
        padding-left: 30px;
        padding-right: 30px;
      }
      .profileImage {
        width: 75px;
        height: 75px;
        border-radius: 50%;
      }
      h4 {
        font-weight: bold;
        font-size: 2em;
        line-height: 38px;
        margin-top: 0px;
      }
      .failure{
        h1{
          border-bottom: none;
          font-weight:900;
          font-size:100px;
          margin:20px 0px;
          padding:0px;
          color:#B2B2B2;
        }
        i{
          font-size:70px;
          color:#EF5350;
        }
        h2{
          font-weight:bold;
          font-size:30px;
        }
      }
    }
    @media only screen and (min-width: 993px) {
      .hide-on-small-only {
        display: block;
      }
      .mainContainer {
        min-height: 800px;
        padding-left: 70px;
        padding-right: 70px;
      }
    }
  }
}

.login-user {
  border: none;
  cursor: pointer;
  color: #3b8ca1;
}
@media screen and (max-width: 991px) {
  .MainWrapper2 .container a.btn.waves-effect.waves-light.editButton {
    position: relative;
    right: 0;
    bottom: 0;
    margin-top: 10%;
  }
}

.report-box {
  padding-top: 40px;
  width: 450px;
  height: 250px;
}

.report-box textarea {
  width: 100%;
  height: 11rem;
  border: #E5E5E5 solid 1px;
  padding: 13px;
  font-size: 16px;
  resize: none;
  border-radius: 3px;
}

.report-box textarea::placeholder {
  color: #C4C4C4;
  opacity: 1;
  font-size: 18px;
}

.report-button {
  display: flex;
  justify-content: flex-end
}

.report-button button {
  border: 1px solid #0e718c;
  color: #0e718c;
  font-size: 15px;
  background-color: #fff;
  padding: 8px 8px;
  outline: none;
  width: 150px;
  cursor: pointer;
  margin-top: 20px;
}


