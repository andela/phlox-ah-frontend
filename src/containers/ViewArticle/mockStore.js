export const articleStore = {
  article: {
    loading: true,
    success: true,
    failure: false,
    article: {
      title: 'new test article by udochukwu',
      body: 'Body here',
      description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. ',
      imgUrl: 'https://images.pexels.com/photos/788485/pexels-photo-788485.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
      readTime: 1,
      createdAt: '2018-11-12T18:16:18.630Z',
      updatedAt: '2018-11-12T18:16:18.630Z',
      User: {
        username: 'nnaji',
        Profile: {
          profileImage: 'https://images.pexels.com/photos/1065084/pexels-photo-1065084.jpeg?cs=srgb&dl=beautiful-beauty-brown-eyes-1065084.jpg&fm=jpg'
        }
      },
      Tags: [
        {
          id: 1,
          name: 'tech',
          createdAt: '2018-09-22T23:00:00.000Z',
          updatedAt: '2018-09-22T23:00:00.000Z',
        },
        {
          id: 2,
          name: 'animals',
          createdAt: '2018-09-22T23:00:00.000Z',
          updatedAt: '2018-09-22T23:00:00.000Z',
        }
      ],
      likes: [],
    }
  },
  comments: {
    comment: [
      {
        id: 3,
        comment: 'you be goat oo',
        createdAt: '2018-11-21T14:00:11.810Z',
        updatedAt: '2018-11-21T14:00:11.810Z',
        articleSlug: 'new-test-article-by-udochukwu-af9329ec-e692-4fe9-8af8-d684dab47178',
        userId: 2,
        User: {
          username: 'nelson',
          email: 'nelson.nnaji@andela.com',
          Profile: null
        },
        likes: [],
        Replies: []
      }
    ]
  },
  user: {
    isAuth: true,
    loading: false
  },
  params: {
    articleslug: 'article-slxug'
  },
  bookmark: {
    bookmarks: []
  },
  report: {
    report: {},
    error: null,
    message: '',
    success: false,
    loading: false
  }
};
