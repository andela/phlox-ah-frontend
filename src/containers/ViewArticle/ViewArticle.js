import React, { Component } from 'react';
import Avatar from 'react-avatar';
import PropTypes from 'prop-types';
import Moment from 'moment';
import { Row } from 'react-materialize';
import StarRatings from 'react-star-ratings';

import './ViewArticle.scss';
import { connect } from 'react-redux';
import ReactHtmlParser from 'react-html-parser';
import CommentTextArea from '../../components/CommentTextArea/CommentTextArea';
import CommentButton from '../../components/CommentButton/CommentButton';
import CommentDisplayBox from '../../components/CommentDisplayBox/CommentDisplayBox';
import { createComment, getAllComment } from '../../requests/CommentRequest';
import ArticleTags from '../../components/Tags/ArticleTags';
import { viewArticle, rateArticle } from '../../requests/ArticleRequests';

/**
 * @class ViewAnArticle
 * @extends {Component}
 */
class ViewArticle extends Component {
  /**
   * @constructor function
   * @param {*} props React properties
   */
  constructor() {
    super();
    this.state = {
      success: false, loading: false, failure: false, article: {}, comment: ''
    };

    this.addComment = this.addComment.bind(this);
    this.addRating = this.addRating.bind(this);
    this.handleChange = this.handleChange.bind(this);
  }


  // eslint-disable-next-line require-jsdoc
  addRating(newRating, name) {
    this.props.rateArticle(this.props.match.params.articleslug, newRating);
  }


  /**
   * @description - This method runs whenever the redux state changes
   * @returns {object} state
   * @param {object} props
   * @param {object} state
   */
  static getDerivedStateFromProps(props, state) {
    if (props.success) {
      const { success, loading, article } = props;
      return { success, loading, article };
    }
    if (props.failure) {
      const {
        success, loading, article, failure
      } = props;
      return {
        success, loading, article, failure
      };
    }
    return state;
  }

  /**
   * @description - This method runs whenever the initial component renders
   * @returns {object} state
   * @param {object} props
   */
  componentDidMount() {
    this.props.viewArticle(this.props.match.params.articleslug);
    this.props.getAllComment(this.props.match.params.articleslug);
  }

  /**
   * @description - This method sets the input values
   * @param {object} e
   * @returns {object} void
   * @memberof Comment
   */
  handleChange(e) {
    this.setState({ [e.target.id]: e.target.value });
  }

  /**
    * @description - This method is used to create comment
    * @returns {object} - return payloads
    * @memberof Comment
    */
  addComment() {
    const { comment, article } = this.state;
    this.setState({ comment: '' });
    this.props.createComment(article.slug, comment);
  }

  /**
  * @description - This method displays the login modal
  * @returns {object} null
  * @memberof Header
  */
  signIn() {
    $('#login-modal').modal('open');
  }

  /**
    * @description - This method is used to render comment input for authenticated users
    * @returns {object} - return jsx
    * @memberof Comment
    */
  renderCommentInput() {
    if (this.props.user.isAuth) {
      return (
        <div>
          <CommentTextArea
            handleChange={this.handleChange}
            value={this.state.comment} />
          <CommentButton addComment={this.addComment} />
        </div>);
    }
    return (
      <p>You must logIn to comment on this article. <button className="login-user" onClick={this.signIn}>LogIn</button>
      </p>
    );
  }

  /**
    * @description - This method is used to render list of comments
    * @returns {object} - return jsx
    * @memberof Comment
    */
  renderCommentList() {
    return this.props.comments.map(
      comment => <CommentDisplayBox
        key={comment.id}
        comment={comment}
        />
    );
  }

  /**
   *
   * @returns {jsx} - jsx
   * @memberof ViewArticle
   */
  dummyContent() {
    return (<div className={this.state.failure && 'hide'}>
    <div className="col s12 l6 lighten-5 img-div">
    <div className="dummyPicture"></div>
    </div>
    <div className="col s12 l6 lighten-5">
    <div className="dummyRight"><div></div><div></div><div></div></div>
    </div>
    <div className="col s12 article-body">
    <div className="dummyBody"><div></div><div></div><div></div></div>
    </div>
    </div>);
  }

  /**
   *
   *
   * @param {*} article
   * @returns {jsx} - jsx
   * @memberof ViewArticle
   */
  realContent(article) {
    return (<div>
      <div className="col s12 l6 img-div">
          <img src={article.imgUrl}/>
      </div>
      <div className="col s12 l6">
          <div>
          <h4 className="capitalize">{article.title}</h4>
          <p className="articleDescription">{article.description}</p>
          <Row className="margin-top-10">
              <div className="col s8">
              <Row className="valign-wrapper">
                  <div className="col s4 m3 l4">
                  {
                    !article.User.Profile || !article.User.Profile.profileImage ? <Avatar name={article.User.username} size="75" round={true} />
                      : <img className="profileImage" src={article.User.Profile.profileImage}/>
                  }
                  </div>
                  <div className="col s8 m9 l8">
                  <span className="writer capitalize">{article.User.username}</span><br/>
                  <span className="date-written capitalize">{Moment.duration(article.createdAt, 'hours').humanize() }</span><br/>
                  <span className="readTime">{article.readTime} Minutes Read</span>
                  </div>
              </Row>
              </div>
              <div className="col s4">
              <button
                  className="btn waves-effect waves-light followButton"
                  type="submit"
                  name="action">Follow
              </button>
              </div>
          </Row>
          <div className="center-align activity-icons">
            <div className="col s2"><i className="fas fa-thumbs-up likeButton liked-unliked"></i> {article.likes.length}</div>
            <div className="col s2"><i className="fas fa-thumbs-down dislikeButton"></i> 3</div>
            <div className="col s1"><i className="fas fa-bookmark bookmarkButton"></i></div>
            <div className="col s1"><i className="fas fa-share-alt shareButton"></i></div>
            {(!this.props.user.isAuth
            || article.User.username === this.props.user.username) && <StarRatings
                    rating={article.ratingAverage}
                    starDimension="20px"
                    className="col s4"
                    starSpacing="5px"
                  /> }
                  {(this.props.user.isAuth
                  && article.User.username !== this.props.user.username) && <StarRatings
                    rating={article.ratingAverage}
                    starDimension="20px"
                    starRatedColor="#5e5f63"
                    className="col s4"
                    changeRating={this.addRating}
                    starSpacing="5px"
                    name='rating'
                  />}
              </div>
              <button
                className="btn waves-effect waves-light editButton"
                type="submit"
                name="action">Edit Article
                  </button>
              </div>
          </div>
          <div className="col s12 article-body">
          {
          ReactHtmlParser(article.body)
          }
      </div>
      <div className="col l4 s12 bold tag-div">
          {<a className="red-text"href="#">Report Article</a>}
      </div>
      </div>);
  }

  /**
   *
   * @returns {jsx} - jsx
   * @memberof ViewArticle
   */
  showContent() {
    if (this.success || this.state.article.User) {
      return this.realContent(this.state.article);
    }
    return this.dummyContent();
  }

  /**
   *
   * @description - This method renders the jsx for this component
   * @returns {jsx} - jsx
   */
  render() {
    const { failure } = this.state;
    return (
    <div className="MainWrapper2">
        <div className="container">
        <Row className="containerRow">
            <div className="col s12 mainContainer ">
            <Row>
                {
                    failure
                    && <div className="center failure">
                         <h1><i className="fas fa-exclamation-triangle"></i></h1>
                         <h1>404</h1>
                         <h2 className="capitalize">We couldnâ€™t find this page.</h2>
                      </div>
                }
                { this.showContent() }
            </Row>
            {this.renderCommentInput()}
            {this.renderCommentList()}
          </div>
        </Row>
        </div>
    </div>
    );
  }
}

ViewArticle.propTypes = {
  article: PropTypes.object,
  articleslug: PropTypes.string,
  comments: PropTypes.array,
  createComment: PropTypes.func,
  getAllComment: PropTypes.func.isRequired,
  failure: PropTypes.bool,
  loading: PropTypes.bool,
  match: PropTypes.object,
  rateArticle: PropTypes.func,
  success: PropTypes.bool,
  user: PropTypes.object,
  viewArticle: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  loading: state.article.loading,
  success: state.article.success,
  failure: state.article.failure,
  article: state.article.article,
  comments: state.comments.comment,
  user: state.user
});

export default connect(mapStateToProps, {
  viewArticle, createComment, getAllComment, rateArticle
})(ViewArticle);
