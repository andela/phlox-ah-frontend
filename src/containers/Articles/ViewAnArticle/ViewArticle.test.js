import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

import ViewArticle from './ViewAnArticle';

const mockStore = configureMockStore();
const store = mockStore({
  Articles: {
    loading: false,
    success: false,
    Article: {
      loading: true,
      success: false,
      failure: false,
      article: {
        title: 'new test article by udochukwu',
        body: '<p>It is a long established fact that a\n                                  reader will be distracted\n                                   by the readable content of\n                                   a page when looking at its layout. The point of using Lorem\n                                   Ipsum is that it\n                                   has a more-or-less\n                                   normal distribution of letters, as opposed to using\n                                  Content here, content here, making it look like readable English.\n                                   Many desktop publishing packages and web page\n                                   editors now use Lorem\n                                    Ipsum as their default model text, and a search for lorem\n                                    ipsum will\n                                     uncover many web sites.\n                                  </p> <p>It is a long established fact that a\n                                  reader will be distracted\n                                   by the readable content of\n                                   a page when looking at its layout. The point of using Lorem\n                                   Ipsum is that it\n                                   has a more-or-less\n                                   normal distribution of letters, as opposed to using\n                                  Content here, content here, making it look like readable English.\n                                   Many desktop publishing packages and web page\n                                   editors now use Lorem\n                                    Ipsum as their default model text, and a search for lorem\n                                    ipsum will\n                                     uncover many web sites.\n                                  </p> <p>It is a long established fact that a\n                                  reader will be distracted\n                                   by the readable content of\n                                   a page when looking at its layout. The point of using Lorem\n                                   Ipsum is that it\n                                   has a more-or-less\n                                   normal distribution of letters, as opposed to using\n                                  Content here, content here, making it look like readable English.\n                                   Many desktop publishing packages and web page\n                                   editors now use Lorem\n                                    Ipsum as their default model text, and a search for lorem\n                                    ipsum will\n                                     uncover many web sites.\n                                  </p> <p>It is a long established fact that a\n                                  reader will be distracted\n                                   by the readable content of\n                                   a page when looking at its layout. The point of using Lorem\n                                   Ipsum is that it\n                                   has a more-or-less\n                                   normal distribution of letters, as opposed to using\n                                  Content here, content here, making it look like readable English.\n                                   Many desktop publishing packages and web page\n                                   editors now use Lorem\n                                    Ipsum as their default model text, and a search for lorem\n                                    ipsum will\n                                     uncover many web sites.\n                                  </p> <p>It is a long established fact that a\n                                  reader will be distracted\n                                   by the readable content of\n                                   a page when looking at its layout. The point of using Lorem\n                                   Ipsum is that it\n                                   has a more-or-less\n                                   normal distribution of letters, as opposed to using\n                                  Content here, content here, making it look like readable English.\n                                   Many desktop publishing packages and web page\n                                   editors now use Lorem\n                                    Ipsum as their default model text, and a search for lorem\n                                    ipsum will\n                                     uncover many web sites.\n                                  </p> ',
        description: 'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. ',
        imgUrl: 'https://images.pexels.com/photos/788485/pexels-photo-788485.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
        readTime: 1,
        createdAt: '2018-11-12T18:16:18.630Z',
        updatedAt: '2018-11-12T18:16:18.630Z',
        User: {
          username: 'nnaji',
          Profile: {
            profileImage: 'https://images.pexels.com/photos/1065084/pexels-photo-1065084.jpeg?cs=srgb&dl=beautiful-beauty-brown-eyes-1065084.jpg&fm=jpg'
          }
        },
        Tags: [
          {
            id: 1,
            name: 'tech',
            createdAt: '2018-09-22T23:00:00.000Z',
            updatedAt: '2018-09-22T23:00:00.000Z',
          },
          {
            id: 2,
            name: 'animals',
            createdAt: '2018-09-22T23:00:00.000Z',
            updatedAt: '2018-09-22T23:00:00.000Z',
          }
        ],
        likes: [],
      }
    }
  }
});

let component;
let myComponent;

describe('<ViewArticle/>', () => {
  beforeEach(() => {
    component = shallow(
      <Provider store={store}>
        <ViewArticle />
      </Provider>
    );
    myComponent = component.dive({ context: { store } }).dive();
  });
  it('should render without throwing an error', () => {
    expect(component).toMatchSnapshot();
  });
});
