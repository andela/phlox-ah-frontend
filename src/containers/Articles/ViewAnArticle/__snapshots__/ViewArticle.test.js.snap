// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ViewArticle/> should render without throwing an error 1`] = `<Connect(ViewAnArticle) />`;
